src/operations/resolvers.tsx

```diff
@@ -0,0 +1,135 @@
+import { Resolvers } from '@apollo/client'
+import { v4 } from 'uuid'
+
+import { GET_COLLECTIONS } from './queries/getCollections'
+import type { Bookmark, Collection } from 'types'
+
+export const localResolvers: Resolvers | Resolvers[] = {
+  Query: {
+    collections: (_root, { id }, { cache }) => {
+      // Returns either all collections if no id provided,
+      // found collection if one matches id,
+      // or an empty array
+
+      if (cache === undefined) {
+        const err = new Error('Cache is undefined')
+        return err
+      }
+
+      let collections: Collection[] = []
+
+      const allCollections = cache.readQuery({
+        query: GET_COLLECTIONS,
+      })
+
+      // We don't have support for graphql errors on
+      // the client side, so return an empty array if null
+      if (allCollections === null) {
+        return collections
+      }
+
+      if (id !== undefined) {
+        collections = allCollections.collections.filter((c: Collection) => {
+          return c.id === id
+        })
+      } else {
+        collections = allCollections.collections
+      }
+
+      return collections
+    },
+  },
+  Mutation: {
+    addCollection: (_root, { id, title, bookmarks }, { cache }) => {
+      const previous = cache.readQuery({
+        query: GET_COLLECTIONS,
+      })
+
+      const newCollection = {
+        id: id,
+        title,
+        bookmarks,
+        __typename: 'Collection',
+      }
+
+      const data = {
+        collections:
+          previous.collections.length !== 0
+            ? [...previous.collections, newCollection]
+            : [newCollection],
+      }
+
+      cache.writeQuery({ query: GET_COLLECTIONS, data })
+      return newCollection
+    },
+    removeCollection: (_root, { id }, { cache }) => {
+      cache.modify({
+        fields: {
+          collections(collections: Collection[], { readField }: any) {
```

Comment:
I need to figure out what type `readField` is

Reply:
passing for now but suz found this resource ffr: https://www.apollographql.com/blog/apollo-client/architecture/redux-to-apollo-data-access-patterns/